src/components/ArticleVote.jsx

```diff
@@ -0,0 +1,63 @@
+import IconButton from '@mui/material/IconButton'
+import ThumbUpOffAltIcon from '@mui/icons-material/ThumbUpOffAlt';
+import ThumbUpAltIcon from '@mui/icons-material/ThumbUpAlt';
+import { useState, useEffect } from 'react';
+import {updateArticleVote} from './api'
+
+// TODO: after user login implemented, prevent user voting multiple times across different sessions
+
+function ArticleVote({article}) {
+    
+    console.log(article.voteCount,'--article.voteCount')
+    const [votes, setVotes] = useState(article.voteCount);
+    const [incrVote, setIncrVote] = useState(0);
+    const [errMessage, setErrMessage] = useState('');
+    console.log(votes,'--votes')
+
+    useEffect(() => {
+        if (incrVote !== 0) {
+            updateArticleVote(article.id, incrVote)
+                .then((articleData) => {
+                    setVotes(articleData.votes);
+                })
+                .catch((err) => {
+                    const errMsg = err.msg
+                        ? err.msg
+                        : 'An unknown error occurred';
+                    setErrMessage(errMsg);
+                });
+        }
+    }, [article, incrVote]);
```

Comment:
for voting, we don't really need to wait to get the updated votes from the api, using optimistic rendering would be a better choice here. So we can make the api call in your handleClick instead of putting it in useEffect and re-render the whole component.